src/analysis.ts

```diff
@@ -127,36 +127,73 @@ export async function analyzeFolders(options: FileAnalysisOptions): Promise<File
   return { fileBundle, analysisResults, ...options };
 }
 
-// function mergeBundleResults(bundle: FileBundle, analysisData: IBundleResult, limitToFiles: string[]): FileBundle {
-//   // Determine max suggestion index in our data
-//   const suggestionIndex = Math.max(...Object.keys(bundle.analysisResults.suggestions).map(i => parseInt(i, 10))) || -1;
-
-//   // Addup all new suggestions' indexes
-//   const newSuggestions = moveSuggestionIndexes<Suggestion>(suggestionIndex, analysisData.analysisResults.suggestions);
-//   const suggestions = { ...bundle.analysisResults.suggestions, ...newSuggestions };
-
-//   const newFiles = fromEntries(
-//     Object.entries(analysisData.analysisResults.files).map(([fn, s]) => {
-//       return [fn, moveSuggestionIndexes(suggestionIndex, s)];
-//     }),
-//   );
-//   const files = {
-//     ...omit(bundle.analysisResults.files, limitToFiles),
-//     ...newFiles,
-//   };
-
-//   const analysisResults = {
-//     ...analysisData.analysisResults,
-//     files,
-//     suggestions,
-//   };
-
-//   return {
-//     ...bundle,
-//     ...analysisData,
-//     analysisResults,
-//   };
-// }
+function mergeBundleResults(
+  oldAnalysisResults: AnalysisResult,
+  newAnalysisResults: AnalysisResult,
+  limitToFiles: string[],
+  removedFiles: string[] = [],
+): AnalysisResult {
+  // Start from the new analysis results
+  // For each finding of the old analysis,
+  //  if it's location is not part of the limitToFiles or removedFiles (removedFiles should also be checked against condeFlow),
+  //   append the finding to the new analysis and check if the rule must be added as well
+  const changedFiles = [...limitToFiles, ...removedFiles];
+  const sarifResults = (newAnalysisResults.sarif.runs[0].results || []).filter(res => {
+      // TODO: This should not be necessary in theory but, in case of two identical files,
+      // Bundle Server returns the finding in both files even if limitToFiles only reports one
+      const loc = res.locations && res.locations[0].physicalLocation?.artifactLocation?.uri;
+      return loc && changedFiles.includes(loc);
+  });
+  const sarifRules = newAnalysisResults.sarif.runs[0].tool.driver.rules || [];
+  const oldResults = oldAnalysisResults.sarif.runs[0].results || [];
+  for (let res of oldResults) {
+    const locations: string[] = (res.locations || []).reduce<string[]>(
```

Comment:
Question: Aren't these `reduce`s basically `map().filter()` / `flatMap().filter()`?
No need to change, but might be easier to read.

Reply:
Great suggestion, It didn't occur to me to simplify it that way

Reply:
Bad news, we need to be compatible with Node 10 and flatMap has been introduced in Node 11... I'm reverting the changes.